File: bot-api/dotnet/Robocode.TankRoyale.BotApi/src/events/DeathEvent.cs
using Newtonsoft.Json;

namespace Robocode.TankRoyale.BotApi.Events;

/// <summary>
/// Event occurring when your bot has died.
/// </summary>
public sealed class DeathEvent : BotEvent
{
    /// <summary>
    /// Indicates if this event is critical, and hence should not be removed from event queue when it gets old.
    /// This event is critical.
    /// </summary>
    /// <return><c>true</c></return>
    public override bool IsCritical => true;

    /// <summary>Initializes a new instance of the DeathEvent class.</summary>
    /// <param name="turnNumber">Turn number when your bot died.</param>
    [JsonConstructor]
    public DeathEvent(int turnNumber) : base(turnNumber)
    {
    }
}

File: bot-api/java/src/main/java/dev/robocode/tankroyale/botapi/events/DeathEvent.java
package dev.robocode.tankroyale.botapi.events;

/**
 * Event occurring when your bot has died.
 */
@SuppressWarnings("unused")
public final class DeathEvent extends BotEvent {

    /**
     * Initializes a new instance of the DeathEvent class.
     *
     * @param turnNumber is the turn number when your bot died.
     */
    public DeathEvent(int turnNumber) {
        super(turnNumber);
    }

    /**
     * {@inheritDoc}
     *
     * <p>This event is critical.
     *
     * @return true
     */
    @Override
    public boolean isCritical() {
        return true;
    }
}